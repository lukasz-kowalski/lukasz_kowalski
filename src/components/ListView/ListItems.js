import React from 'react';
import ListItem from './ListItem';
import ListSortButton from './ListSortButton';

const buttonNames = [
  'Id', 'Name', 'Species', 'Gender', 'Homeworld', 'Actions'
]

const ListItems = ({ characters, deleteCharacter, handleSortChange }) => (
  <table className="table table-bordered table-hover">
    <thead className="thead-light">
    <tr>
      {buttonNames.map(item => 
        <ListSortButton
          key={item}
          handleSortChange={handleSortChange}
          name={item}
        />
      )}
    </tr>
    </thead>
    <tbody>
    {characters.map(character =>
      <ListItem
        key={character.name + character.id}
        character={character} 
        deleteCharacter={deleteCharacter}
      />
    )}
    </tbody>
  </table>
);

export default ListItems;
