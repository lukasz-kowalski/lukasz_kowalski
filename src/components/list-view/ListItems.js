import React from 'react';

const ListItems = ({ characters }) => (
  <table className="table table-bordered table-hover">
    <thead className="thead-light">
    <tr>
      <th scope="col">Id</th>
      <th scope="col">Name</th>
      <th scope="col">Species</th>
      <th scope="col">Gender</th>
      <th scope="col">Homeworld</th>
      <th scope="col">Actions</th>
    </tr>
    </thead>
    <tbody>
    {characters.map((character, index) => 
      <tr key={character.name}>
        <th scope="row">{index + 1}</th>
        <td>{character.name}</td>
        <td>{character.species}</td>
        <td>{character.gender}</td>
        <td>{character.homewordl}</td>
        <td>
          <div className="btn-group btn-group-sm" role="group" aria-label="Actions">
            <button type="button" className="btn btn-secondary">
              <i className="fa fa-pencil" aria-hidden="true"/> Edit
            </button>
            <button type="button" className="btn btn-danger">
              <i className="fa fa-trash-o" aria-hidden="true"/> Remove
            </button>
          </div>
        </td>
      </tr>
    )}
    </tbody>
  </table>
);

export default ListItems;
