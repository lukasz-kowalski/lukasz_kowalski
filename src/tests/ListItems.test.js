import React from 'react';
import { shallow } from 'enzyme';
import ListItems from '../components/ListView/ListItems';

let wrapper;

const characters = [
  {
    "id": 1,
    "name": "Luke Skywalker",
    "species": "Human",
    "gender": "male",
    "homeworld": "Tatooine"
  },
  {
    "id": 2,
    "name": "C-3PO",
    "species": "Droid",
    "gender": "n/a",
    "homeworld": "Tatooine"
  }
];

beforeEach(() => {
  wrapper = shallow(
  <ListItems
    characters={characters}
  />);
})

test('should successful render ListItems', () => {
  expect(wrapper).toMatchSnapshot();
})
