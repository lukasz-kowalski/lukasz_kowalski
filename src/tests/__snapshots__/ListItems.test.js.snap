// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successful render ListItems 1`] = `
<table
  className="table table-bordered table-hover"
>
  <thead
    className="thead-light"
  >
    <tr>
      <ListSortButton
        key="Id"
        name="Id"
      />
      <ListSortButton
        key="Name"
        name="Name"
      />
      <ListSortButton
        key="Species"
        name="Species"
      />
      <ListSortButton
        key="Gender"
        name="Gender"
      />
      <ListSortButton
        key="Homeworld"
        name="Homeworld"
      />
      <ListSortButton
        key="Actions"
        name="Actions"
      />
    </tr>
  </thead>
  <tbody>
    <ListItem
      character={
        Object {
          "gender": "male",
          "homeworld": "Tatooine",
          "id": 1,
          "name": "Luke Skywalker",
          "species": "Human",
        }
      }
      key="Luke Skywalker1"
    />
    <ListItem
      character={
        Object {
          "gender": "n/a",
          "homeworld": "Tatooine",
          "id": 2,
          "name": "C-3PO",
          "species": "Droid",
        }
      }
      key="C-3PO2"
    />
  </tbody>
</table>
`;
